packages/quantic/force-app/main/default/lwc/quanticDidYouMean/__tests__/quanticDidYouMean.test.js

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable no-import-assign */
+// @ts-ignore
+import {createElement} from 'lwc';
+import QuanticDidYouMean from '../quanticDidYouMean';
+import * as mockHeadlessLoader from 'c/quanticHeadlessLoader';
+
+jest.mock('c/quanticHeadlessLoader');
+
+let isInitialized = false;
+
+const exampleEngine = {
+  id: 'exampleEngineId',
+};
+
+const defaultQueryCorrectionMode = 'legacy';
+
+const defaultOptions = {
+  engineId: exampleEngine.id,
+  disableQueryAutoCorrection: false,
+  queryCorrectionMode: defaultQueryCorrectionMode,
+};
+
+const mockDidYouMeanState = {
+  wasCorrectedTo: 'example corrected query',
+  originalQuery: 'example original query',
+  wasAutomaticallyCorrected: true,
+  queryCorrection: {
+    correctedQuery: 'example corrected query',
+    wordCorrected: 'example',
+  },
+  hasQueryCorrection: true,
+};
+
+const mockQueryTriggerState = {
+  newQuery: 'example new query',
+  originalQuery: 'example original query',
+  wasQueryModified: true,
+};
```

Comment:
Why exactly do you need to mock these?
I would do all we can to keep our test setup minimal, and these I don't think with the tests you are doing are really necessary,

Reply:
we dont need the `mockQueryTriggerState` per se, but we do need the `mockDidYouMeanState` for its queryCorrection object.

otherwise the tests would fail with the following error:
<img width="658" alt="image" src="https://github.com/user-attachments/assets/a70a34f0-50d9-4cee-9499-f160695deb18">

they dont need the `mockDidYouMeanState` to have all the values just the queryCorrection, but at this point might as well show the entire state object.